### PR TITLE
Hs 3012 as

### PR DESCRIPTION
**Pull request by aslobodeniuk**

### CODE REVIEW

File: sys/androidmedia/gstamc.h [hunk 1]
```diff
@@ -83,14 +83,18 @@ struct _GstAmcCrypto
 
 struct _GstAmcCodec
 {
+  guint flush_id;
+  GMutex buffers_lock;
   /* < private > */
   jobject object;               /* global reference */
+  volatile gint ref_count;
```
Comment (turran) [inline sys/androidmedia/gstamc.h]: This is not needed. No code is using it but something added by yourself. Please maintain implicit code guidelines.

File: sys/androidmedia/gstamc.c [hunk 2]
```diff
@@ -653,6 +678,159 @@ gst_amc_codec_get_release_method_id (GstAmcCodec * codec)
   return media_codec.release_output_buffer;
 }
 
+
```
Comment (turran) [inline sys/androidmedia/gstamc.c:696]: The method id should be cached as any other method already cached in get_java_classes

**Review by turran — COMMENTED** [review]
Also after the corresponding fixups, do not forget to squash and rename the commit messages accordingly

File: sys/androidmedia/gstamc.h (hunk 1 above)
Comment (aslobodeniuk) [reply sys/androidmedia/gstamc.h]: gint
g_atomic_int_add (volatile gint *atomic,
                  gint val);

Comment (turran) [reply sys/androidmedia/gstamc.h]: That's the method definition. Is a parameter qualifier. The member of the struct/class does not need to be volatile. Is just a compiler hint when that method is called.

File: sys/androidmedia/gstamc.c (hunk 2 above)
Comment (aslobodeniuk) [reply sys/androidmedia/gstamc.c:696]: catching done

File: sys/androidmedia/gstamc.h (hunk 1 above)
Comment (aslobodeniuk) [reply sys/androidmedia/gstamc.h]: ok, done

**Review by ylatuya — APPROVED** [review]

**Pull request merged**